salt/modules/zypper.py

```diff
@@ -1768,17 +1820,53 @@ def search(criteria, refresh=False):
     if refresh:
         refresh_db()
 
-    solvables = __zypper__.nolock.xml.call('se', criteria).getElementsByTagName('solvable')
+    cmd = ['search']
+    if kwargs.get('match') == 'exact':
+        cmd.append('--match-exact')
+    elif kwargs.get('match') == 'words':
+        cmd.append('--match-words')
+    elif kwargs.get('match') == 'substrings':
+        cmd.append('--match-substrings')
+
+    if kwargs.get('provides'):
+        cmd.append('--provides')
+    if kwargs.get('recommends'):
+        cmd.append('--recommends')
+    if kwargs.get('requires'):
+        cmd.append('--requires')
+    if kwargs.get('suggests'):
+        cmd.append('--suggests')
+    if kwargs.get('conflicts'):
+        cmd.append('--conflicts')
+    if kwargs.get('obsoletes'):
+        cmd.append('--obsoletes')
+
+    if kwargs.get('file_list'):
+        cmd.append('--file-list')
+    if kwargs.get('search_descriptions'):
+        cmd.append('--search-descriptions')
+    if kwargs.get('case_sensitive'):
+        cmd.append('--case-sensitive')
+    if kwargs.get('installed_only'):
+        cmd.append('--installed-only')
+    if kwargs.get('not_installed_only'):
+        # long parameter was renamed in the past
+        cmd.append('-u')
+    if kwargs.get('details'):
+        cmd.append('--details')
```

Comment:
I still don't like this. :wink: `file_list` is `--file-list`, `details` are just `--details`, `installed_only` is `--installed-only` and so on. This all can be set just in one loop instead of `0xFF` `if/else` clauses. And if there is a Zypper with a newer option that can be _just passed_, it could be _just passed_.

Let's change this, please.